src/api/JobUtils.ts

```diff
@@ -75,17 +75,20 @@ export class JobUtils {
     public static async getSpoolFiles(connection: any, jobId: string): Promise<ISpoolFile[]> {
         const jobDetails = (await JobUtils.findJobByID(connection, jobId));
         const fullSpoolFiles: ISpoolFile[] = [];
-        for (const spoolFileToDownload of jobDetails.spoolFiles) {
-            this.log.debug("Requesting spool files for job %s(%s) spool file ID %d", jobDetails.jobname, jobDetails.jobid, spoolFileToDownload.id);
-            const option = {
-                jobName: jobDetails.jobname,
-                jobId: jobDetails.jobid,
-                owner: "*",
-                fileId: spoolFileToDownload.id
-            };
-            const spoolFile = await JobUtils.getSpoolFileContent(connection, option);
-            spoolFileToDownload.contents = spoolFile;
-            fullSpoolFiles.push(spoolFileToDownload);
+        if (jobDetails.spoolFiles) {
+            for (const spoolFileToDownload of jobDetails.spoolFiles) {
```

Comment:
Wondering if we should push the condition down to the `for-of` loop. 
For example `for (const spoolFileToDownload of jobDetails.spoolFiles ?? []) {`